### PR TITLE
Add profile assertions

### LINKED ISSUE #34: Profile Test Assertions
Add a document that defines the test assertions required for the Catena-X Profile TCK.

### PR DESCRIPTION
**Pull request by jimmarino**
Add an assertions for the Catena-X Profile TCK.

Closes #34.

### CODE REVIEW

File: cx/tck/tck.assertions.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+### 1. 3. Negative Expired Membership Credential
+
+The test client will request a DSP catalog from the provider using the DSP/DCP protocol combination and provide an
+expired `Membership Credential`.
+
+#### Assertion
+
+The assertion will verify that an `HTTP 4XX` error is returned.
+
+### 1. 3. Negative Invalid DID Membership Credential
```
Comment (paullatzelsperger) [inline cx/tck/tck.assertions.md]:
```suggestion
### 1. 4. Negative Invalid DID Membership Credential
```

File: cx/tck/tck.assertions.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+The test client will request a DSP catalog from the provider using the DSP/DCP protocol combination and provide
+a `Membership Credential` with an invalid `DID` for the `id` entry.
+
+#### Assertion
+
+The assertion will verify that an `HTTP 4XX` error is returned.
+
+### 1. 4. Negative Invalid Holder Identifier Membership Credential
```
Comment (paullatzelsperger) [inline cx/tck/tck.assertions.md]:
```suggestion
### 1. 5. Negative Invalid Holder Identifier Membership Credential
```

File: cx/tck/tck.assertions.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+For each framework agreement subtype, the test client will request a DSP catalog from the provider using the DSP/DCP
+protocol combination and provide a valid Framework `Credential Credential` subtype.
+
+#### Assertion
+
+The assertion will verify that a valid `DCAT 3` catalog is returned containing an `Offer` for the dataset corresponding
+to each framework agreement subtype.
+
+### 3. 2. Negative No Dismantler Credential
```
Comment (paullatzelsperger) [inline cx/tck/tck.assertions.md]:
```suggestion
### 3. 2. Negative No Framework Credential
```

File: cx/tck/tck.assertions.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+The test client will request a DSP catalog from the provider using the DSP/DCP protocol combination and not provide
+a `Framework Credential`.
+
+#### Assertion
+
+The assertion will verify that an `HTTP 4XX` error is returned.
+
+### 3. 3. Negative Expired Dismantler Credential
```
Comment (paullatzelsperger) [inline cx/tck/tck.assertions.md]:
```suggestion
### 3. 3. Negative Expired Framework Credential
```

File: cx/tck/tck.assertions.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+The test client will request a DSP catalog from the provider using the DSP/DCP protocol combination and provide an
+expired `Framework Credential`.
+
+#### Assertion
+
+The assertion will verify that an `HTTP 4XX` error is returned.
+
+### 3. 3. Negative Invalid DID Dismantler Credential
```
Comment (paullatzelsperger) [inline cx/tck/tck.assertions.md]:
```suggestion
### 3. 3. Negative Invalid DID Framework Credential
```

File: cx/tck/tck.assertions.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+The test client will request a DSP catalog from the provider using the DSP/DCP protocol combination and provide
+a `Framework Credential` with an invalid `DID` for the `id` entry.
+
+#### Assertion
+
+The assertion will verify that an `HTTP 4XX` error is returned.
+
+### 3. 4. Negative Invalid Holder Identifier Dismantler Credential
```
Comment (paullatzelsperger) [inline cx/tck/tck.assertions.md]:
```suggestion
### 3. 4. Negative Invalid Holder Identifier Framework Credential
```

**Review by paullatzelsperger — APPROVED** [review]
some copy-paste errors

**Review by wolf4ood — APPROVED** [review]

**Pull request merged**